presto-hive/src/main/java/com/facebook/presto/hive/RecordFileWriter.java

```diff
@@ -160,7 +162,13 @@ public void commit()
     public void rollback()
     {
         try {
-            recordWriter.close(true);
+            try {
+                recordWriter.close(true);
+            }
+            finally {
+                // perform explicit delete of written file here as recordWriter.close() ignores the abort flag.
```

Comment:
perform explicit deletion here as implementations of RecordWriter.close() often ignore the abort flag.